tools/setup_helpers/cmake.py

```diff
@@ -287,6 +285,16 @@ def generate(
             elif var.startswith(('BUILD_', 'USE_', 'CMAKE_')):
                 build_options[var] = val
 
+        # The default value cannot be easily obtained in CMakeLists.txt. We set it here.
+        py_lib_path = sysconfig.get_path('purelib')
+        cmake_prefix_path = build_options.get('CMAKE_PREFIX_PATH', None)
+        if cmake_prefix_path:
+            build_options["CMAKE_PREFIX_PATH"] = (
+                cast(str, py_lib_path) + ";" + cast(str, cmake_prefix_path)
+            )
```

Comment:
Nit: please use `f` strings, or `'{};{}'.format(py_lib_path, cmake_prefix_path)`, which would likely eliminate the need of cast (as it will call `.__repr__` method, isn't it?)
```suggestion
        py_lib_path = cast(str, sysconfig.get_path('purelib'))
        cmake_prefix_path = cast(str, build_options.get('CMAKE_PREFIX_PATH', None))
        if cmake_prefix_path:
            build_options["CMAKE_PREFIX_PATH"] = f'{py_lib_path};{cmake_prefix_path}'
```

Reply:
Sure, I can do that. I wasn't sure whether our build scripts were supposed to support earlier versions that Python v3.6. I saw a few places in the code where we concatenate things, so didn't want to risk it.